src/FileSystem-Disk/ReadOnlyMacStore.class.st

```diff
@@ -0,0 +1,102 @@
+"
+I'm a specific store for macOS file systems that allows only read-only access. All read-only DiskStore subclasses share the same behavior. Usage of traits would be appropriate here, but it is not used because the kernel should not contain traits.
+"
+Class {
+	#name : #ReadOnlyMacStore,
+	#superclass : #MacStore,
+	#category : #'FileSystem-Disk-Store'
+}
+
+{ #category : #public }
+ReadOnlyMacStore class >> isActiveClass [
+
+	^ false
+]
+
+{ #category : #public }
+ReadOnlyMacStore class >> writableVariant [
+
+	^ MacStore
+]
+
+{ #category : #public }
+ReadOnlyMacStore >> basicOpen: aPath writable: aBoolean [
+
+	aBoolean ifTrue: [ 
+		(ReadOnlyFileException new
```

Comment:
This can be replaced with something like:
```
ReadOnlyFileException signal: 'Attempt to open file ' , aPath pathString, ' as writable on a read-only file system'.
```
By default sending signal: to an exception class with set the messageText

Reply:
done